src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,258 @@ begin
+begin
+  apply induction_on f,
+  { intro a,
+    simp },
+  { intros p q hp hq,
+    simp only [(fin_succ_equiv R n).map_add, ring_hom.map_add, polynomial.map_add,
+              polynomial.eval_add],
+    congr,
+    { exact hp },
+    { exact hq } },
+  { intros p j h,
+    simp only [(fin_succ_equiv R n).map_mul, eval_X, polynomial.map_mul, ring_hom.map_mul,
+              polynomial.eval_mul],
+    congr,
+    { exact h },
+    { clear h f,
+      simp only [fin_succ_equiv_apply, eval₂_hom_X'],
+      by_cases c : j = 0,
+      { rw c,
+        simp [fin.cons_zero] },
+      { have c' : j ≠ 0 := by simpa only [ne.def],
+        rw [←fin.succ_pred j c', fin.cons_succ],
+        simp } } },
+end
+
+lemma coeff_eval_eq_eval_coeff {n : ℕ} (s' : fin n → R) (f : polynomial (mv_polynomial (fin n) R))
+  (i : ℕ) : polynomial.coeff (polynomial.map (eval s') f) i =  eval s' (polynomial.coeff f i) :=
+by simp only [polynomial.coeff_map]
+
+/- TODO: what is a good name for this one? -/
+lemma support_eval {n : ℕ} (s' : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) : (polynomial.map (eval s') f).support ⊆ f.support :=
+begin
+  intros i hi,
+  simp only [polynomial.mem_support_iff, polynomial.coeff_map, ne.def] at hi,
+  by_contradiction c,
+  simp only [polynomial.mem_support_iff, not_not, ne.def] at c,
+  rw c at hi,
+  simpa using hi,
+end
+
+lemma degree_eval_le_degree {n : ℕ} (s' : fin n → R) (f : polynomial (mv_polynomial (fin n) R)) :
+  polynomial.degree (polynomial.map (eval s') f) ≤ polynomial.degree f :=
+polynomial.degree_mono (support_eval s' f)
+
+lemma nat_degree_eval_le_nat_degree {n : ℕ} (s : fin n → R)
+  (f : polynomial (mv_polynomial (fin n) R)) :
+  polynomial.nat_degree (polynomial.map (eval s) f) ≤ polynomial.nat_degree f :=
+polynomial.nat_degree_le_nat_degree (degree_eval_le_degree s f)
+
+lemma support_coeff_fin_succ_equiv {n : ℕ} {f : mv_polynomial (fin (n + 1)) R} {i : ℕ}
+  {m : fin n →₀ ℕ } : m ∈ (polynomial.coeff ((fin_succ_equiv R n) f) i).support
+   ↔ (finsupp.cons i m) ∈ f.support :=
+begin
+  apply iff.intro,
+  { intro h,
+    simpa [←fin_succ_equiv_coeff_coeff] using h },
+  { intro h,
+    simpa [mem_support_iff, ←fin_succ_equiv_coeff_coeff m f i] using h },
+end
+
+lemma fin_succ_equiv_support {n : ℕ} (f : mv_polynomial (fin (n + 1)) R) :
+  (fin_succ_equiv R n f).support = finset.image (λ m : fin (n + 1)→₀ ℕ, m 0) f.support :=
+begin
+  apply finset.subset.antisymm,
+  { intros i hi,
+    rw finset.mem_image,
+    rw [polynomial.mem_support_iff, nonzero_iff_exists] at hi,
+    cases hi with m hm,
+    use cons i m,
+    apply and.intro,
+    { rw ← support_coeff_fin_succ_equiv,
+      simpa using hm },
+    { rw cons_zero } },
+  { intros i hi,
+    rw polynomial.mem_support_iff,
+    rw finset.mem_image at hi,
+    cases hi with m hm,
+    cases hm with h hm,
+    rw nonzero_iff_exists,
+    use tail m,
+    rwa [← coeff, ← mem_support_iff, support_coeff_fin_succ_equiv, ← hm, cons_tail] },
+end
```

Comment:
A much shorter proof. If you find yourself rewriting by the same lemma in both goals, you probably should have done the rewrite before you split the goal:
```suggestion
lemma fin_succ_equiv_support {n : ℕ} (f : mv_polynomial (fin (n + 1)) R) :
  (fin_succ_equiv R n f).support = finset.image (λ m : fin (n + 1)→₀ ℕ, m 0) f.support :=
begin
  ext i,
  rw [polynomial.mem_support_iff, finset.mem_image, nonzero_iff_exists],
  split,
  { rintro ⟨m, hm⟩,
    refine ⟨cons i m, _, cons_zero _ _⟩,
    { rw ← support_coeff_fin_succ_equiv,
      simpa using hm }, },
  { rintro ⟨m, h, rfl⟩,
    refine ⟨tail m, _⟩,
    rwa [← coeff, ← mem_support_iff, support_coeff_fin_succ_equiv, cons_tail] },
end
```